numpy/lib/function_base.py

```diff
@@ -1912,10 +1911,16 @@ def diff(a, n=1, axis=-1):
     slice2[axis] = slice(None, -1)
     slice1 = tuple(slice1)
     slice2 = tuple(slice2)
+
+    if a.dtype == np.bool_:
+        da = a[slice1] ^ a[slice2]
```

Comment:
>we can add support for dtypes with fields here too
@eric-wieser I remember that amazing feature I was always willing to try :). Is better to keep it for a separate PR though.